.github/workflows/publish-package.yaml

```diff
@@ -1,38 +1,50 @@
-name: Publish Package
+name: Tag & Publish package
 
 on:
-  push:
-    tags:
-      - 'barcode-*'
-      - 'channable-*'
-      - 'clockify-enhanced-*'
-      - 'cronhooks-*'
-      - 'fulfillmenttools-*'
-      - 'google-enhanced-*'
-      - 'kaufland-marketplace-*'
-      - 'moco-*'
-      - 'otto-market-*'
-      - 'sentry-io-enhanced-*'
+  pull_request:
+    branches:
+      - main
+    types:
+      - closed
+
+permissions:
+  contents: write
 
 env:
   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-  NODE_VERSION: 18.19.1
+  NODE_VERSION: 18.20.2
 
 jobs:
-  publish-package:
-    name: Publish Package
+  tag-and-publish-package:
+    name: Tag & Publish
     runs-on: ubuntu-latest
+    # Only run this job if an actual publishing should be made
+    if: ${{ (github.event.pull_request.merged == true) &&
+      (startsWith(github.event.pull_request.head.ref, 'release/')) &&
+      (contains(github.event.pull_request.labels.*.name, 'release')) }}
 
     steps:
+      - name: Extract package from head ref
+        id: extract_package
+        # yamllint disable-line rule:line-length
+        run: echo "package=$(echo ${{ github.event.pull_request.head.ref }} | grep -oP '(?<=release/)[^@]+(?=@)')" >> "$GITHUB_OUTPUT"
+
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          filter: 'blob:none'
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Configure git User
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email '41898282+github-actions[bot]@users.noreply.github.com'
 
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
```

Comment:
Added, but I have removed the "Restore Node.js modules cache" step based on [this](https://github.com/actions/cache/blob/main/examples.md#node---npm) recommendation from the `GitHub Cache Action`

> Note It is not recommended to cache node_modules, as it can break across Node versions and won't work with npm ci